### PR TITLE
Avoid adding an extra newline after block comments

### PR DESCRIPTION
**Pull request by Marwes**
When block comments were rewritten to line comments they check if a new line needs to be added and adds one if needed. It only checked for '\n' however which would cause a newline to be added even if the comment was ended by "\r\n".

### CODE REVIEW

**Comment by Marwes** [comment]
Let me know if I need to change something.

File: src/missed_spans.rs [hunk 1]
```diff
@@ -143,11 +143,12 @@ impl<'a> FmtVisitor<'a> {
...
+                        // Add a newline after line comments
                         self.buffer.push_str("\n");
                     } else if line_start < snippet.len() {
-                        let x = (&snippet[line_start..]).chars().next().unwrap() != '\n';
-
-                        if x {
+                        // For other comments add a newline if there isn't one at the end already
+                        let c = snippet[line_start..].chars().next().unwrap();
+                        if c != '\n' && c != '\r' {
                             self.buffer.push_str("\n");
```
Comment (nrc) [inline src/missed_spans.rs]: Here and above, I think we should be checking the option for what kind of newline to append, rather than just adding `/n`. I think there should be some facility for doing this (possibly with some refactoring required) in https://github.com/rust-lang-nursery/rustfmt/blob/master/src/filemap.rs

Comment (Marwes) [inline src/missed_spans.rs]: Currently it seems that `\n` is always used as newlines and it is only when writing to file/stdout etc that the actual line ending is decided on. https://github.com/rust-lang-nursery/rustfmt/blob/master/src/filemap.rs#L75-L80

It might be a good idea to always use the correct line but that should probably be a separate PR (?)

Comment (nrc) [inline src/missed_spans.rs]: ah ok, sounds good.

**Comment by nrc** [comment]
Thanks for the PR!
